### DISCUSSION TITLE
Benchmark assertions

### DISCUSSION THREAD

**Discussion by f15u** [vitest-dev/vitest#4225] [Q&A]
In #1029, the `bench` feature was introduced. After reviewing https://github.com/vitest-dev/vitest/issues/917#issuecomment-1079624003, it appears that there was a proposal to retrieve benchmark values and perform assertions on them. Has this proposal been abandoned during implementation? If not, is it possible to make the benchmark fail if the specified criteria are not met?

For example, we have implemented a search engine in our application layer that needs to be tested, and it should return results in less than 1 second.

Currently, we have a test that fails if the `p99` value from `tinybench` exceeds 1000 milliseconds. Having the benchmark directly integrated into Vitest reduces boilerplate, maintains consistency in our test suite, and provides us with the opportunity to run these tests in parallel in our CI pipeline.

**Comment by sheremet-va** [comment]
This would be a great feature. I don't think there is anyone on the team who is working on benchmarking right now, but if you open a separate issue, I am sure someone might pick it up.
